providers/base/bin/prime_offload_tester.py

```diff
@@ -44,54 +44,70 @@ class PrimeOffloader:
     logger = logging.getLogger()
     check_result = False
 
-    def find_card_id(self, pci_name: str) -> str:
+    def _run_command(self, cmd: list, shell=False) -> str:
+        """
+        use subprocess.check_output to execute command
+
+        :param cmd: the command will be executed
+
+        :param shell: enbale shell or not
+
+        :returns: Ouput of command
+        """
+        try:
+            return subprocess.check_output(
+                cmd, shell=shell, universal_newlines=True
+            )
+
+        except (subprocess.CalledProcessError, FileNotFoundError) as e:
+            raise SystemExit(
+                "Running command:{} failed due to {}".format(cmd, repr(e))
+            )
+
+    def find_card_id(self, pci_bdf: str) -> str:
         """
-        use pci name to find card id under /sys/kernel/debug/dri
+        use pci BDF to find card id under /sys/kernel/debug/dri
 
-        :param pci_name: pci device name in NNNN:NN:NN.N format
+        :param pci_bdf: pci device BDF in NNNN:NN:NN.N format
 
         :returns: card id
         """
-        pci_name_format = "[0-9]{4}:[0-9,a-f]{2}:[0-9,a-f]{2}.[0-9]"
-        if not re.match(pci_name_format, pci_name.lower()):
-            raise SystemExit("pci name format error")
+        pci_bdf_format = "[0-9]{4}:[0-9,a-f]{2}:[0-9,a-f]{2}.[0-9]"
+        if not re.match(pci_bdf_format, pci_bdf.lower()):
+            raise SystemExit("pci BDF format error")
 
         try:
             cmd = [
                 "grep",
                 "-lr",
                 "--include=name",
-                pci_name,
+                pci_bdf,
                 "/sys/kernel/debug/dri",
             ]
```

Comment:
Can't you perform the same function without using grep? Something like:
```
directory = "/sys/kernel/debug/dri"
file_pattern = "name"
search_string = "pci_name"

for root, dirs, files in os.walk(directory):
    for file_name in fnmatch.filter(files, file_pattern):
        file_path = os.path.join(root, file_name)
        # Check if the search string is in the file
        with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
            if search_string in file.read():
                return file_path
```

That way you could also reuse the find function for `find_offload()`